vsintegration/src/FSharp.Editor/Classification/ClassificationService.fs

```diff
@@ -20,7 +22,42 @@ open Microsoft.CodeAnalysis.ExternalAccess.FSharp.Classification
 // IVT, we'll maintain the status quo.
 #nowarn "44"
 
+#nowarn "57"
+
+open FSharp.Compiler.Range
 open FSharp.Compiler.SourceCodeServices
+open FSharp.Compiler.SourceCodeServices.Lexer
+
+type SemanticClassificationData = (struct(FSharp.Compiler.Range.range * SemanticClassificationType)[])
+type SemanticClassificationLookup = IReadOnlyDictionary<int, ResizeArray<struct(FSharp.Compiler.Range.range * SemanticClassificationType)>>
+
+[<Sealed>]
+type DocumentCache<'Value when 'Value : not struct>() =
+    let cache = new MemoryCache("fsharp-cache")
+    let policy = CacheItemPolicy(SlidingExpiration = TimeSpan.FromSeconds 2.)
```

Comment:
Why 2.0 seconds here?

Reply:
Anything under 2 seconds, I believe, the caching stops working, meaning it won't actually cache the item. It's a really stupid bug. So, 2 seconds is really the minimum that we can go IIRC.

Reply:
Can this be added as a comment?

Reply:
Yea, makes sense.

Reply:
Can we make it tunable via an environment variable, like a lot of other such settings? I always think it's good practice in case we have to have customers do in situ testing of a different setting?

Reply:
We could expose this as a setting, but I really think we should not. What we have should just work without any tweaking. Adding more time to this could make it worse; remember if the we cache the same item again it will reset the sliding expiration time back to 0.

Reply:
I don't think this should be a setting.